### PR TITLE
Feature: Add steps for getting ACR credentials

### PR DESCRIPTION
**Pull request by albertdb**
@sundharakanth please review and test. Try to fix spellcheck issues which are not really spelling mistakes.

### CODE REVIEW

**Comment by sundharakanth** [comment]
To resolve the errors from spellcheck, we need to add the 6 valid words (ECR, ecr, dkr, aws, amazonaws, azurecr) in the dictionary - https://github.com/devonfw/.github/blob/master/en-devonfw.dict
image

There is one word mis-spelt as "accesing" instead of "accessing" which is to be corrected.

File: documentation/setup-container-image-registry.asciidoc [hunk 1]
```diff
@@ -134,6 +134,22 @@ That is:
...
+* Retrieve credentials for accesing the registry with the following command:
+
+[source,shell]
+----
+   az acr credential show --name <registry-name>
+----
+
 === Image Format
 * `<registry-url>/<namespace>/<image-name>:<tag>`
 ** `<registry-url>`: <registry-name>.azurecr.io
```
Comment (sundharakanth) [inline documentation/setup-container-image-registry.asciidoc:155]: ** `<registry-url>`: Format is `<registry-name>.azurecr.io`

**Review by sundharakanth — CHANGES_REQUESTED** [review]
Please change line#83 from 
** `<registry-url>`: <aws-account-id>.dkr.ecr.<region>.amazonaws.com
to
** `<registry-url>`: Format is `<aws-account-id>.dkr.ecr.<region>.amazonaws.com`

We can add only new technical words into dictionary and to avoid adding words like "dkr" and "ecr", we will have to show the above line as code.

**Comment by sundharakanth** [comment]
Hi @albertdb - Opened a PR https://github.com/devonfw/.github/pull/126 to add new words ECR and ACR in Dictionary and got it reviewed and merged. For all other words failing, we need to use backticks (as code) since those are not new technical words rather those are specific words from each providers as per Devon Core Team review.

**Comment by albertdb** [comment]
@sundharakanth Thanks. Apart from that, do the commands work as expected?

**Comment by sundharakanth** [comment]
> @sundharakanth Thanks. Apart from that, do the commands work as expected?

Thanks for the updates @albertdb . Yes, all the commands are working as expected and once we merge the current file, we can check if it resolves SpellCheck issues.

**Comment by sundharakanth** [comment]
Hi @albertdb , right now, only the 3 words (imagesdir, partnums, sectnums) are failing in SpellCheck. 

image

There are a couple of options to fix this issue:
1. We can check with Jan @SchettlerKoehler if we are going to use the below format in all the documentation folders as template in all devonfw github projects and if not, we can remove this part.

image

2. If this format is required, then, we can check with Malte @maybeec and add regex to escape words starting with colon and ending with colon to escape words like 
          (a) :imagesdir:
          (b) :partnums:
          (c) :sectnums:

image

**Comment by sundharakanth** [comment]
> @sundharakanth Please go ahead preparing the solution mentioned in the second option while we get more opinions on this matter. Not much work to throw away in case a different decision is finally taken.

@albertdb - The following regex takes care of escaping the single words (not the ones with hyphen in between like :toc-title: ). Please confirm. 
image

**Comment by sundharakanth** [comment]
Hi All ( @maybeec , @SchettlerKoehler , @ssarmokadam , @isandesh1986 , @sujith-mn ),
I can see the following templated related non-standard words in dictionary.

148   doctype
614   toc
249   idprefix
250   idseparator
516   reproducable [Looks like mis-spelling of reproducible]

If we have the above words, then, I think, it is okay to add the following few more words. Please confirm. 
(a) :imagesdir:
(b) :partnums:
(c) :sectnums:

image

[ FYI - @albertdb ]

**Comment by albertdb** [comment]
> Hi All ( @maybeec , @SchettlerKoehler , @ssarmokadam , @isandesh1986 , @sujith-mn ), I can see the following templated related non-standard words in dictionary.
> 
> 148 doctype 614 toc 249 idprefix 250 idseparator 516 reproducable [Looks like mis-spelling of reproducible]
> 
> If we have the above words, then, I think, it is okay to add the following few more words. Please confirm. (a) :imagesdir: (b) :partnums: (c) :sectnums:
> 
> image
> 
> [ FYI - @albertdb ]

That is kicking the can down the road, because we can have arbitrary attributes in a document: https://asciidoctor.org/docs/asciidoc-recommended-practices/#document-attributes-i-e-variables



> > @sundharakanth Please go ahead preparing the solution mentioned in the second option while we get more opinions on this matter. Not much work to throw away in case a different decision is finally taken.
> 
> @albertdb - The following regex takes care of escaping the single words like _:sectnums:_ , _:partnums:_ , _:imagesdir:_ , etc. Please confirm. [ FYI - @ssarmokadam , @isandesh1986 , @sujith-mn ]
> 
> image
> 
> **Note:** The above one does not take care of the words with hyphen in between like _:toc-title:_ , _:chapter-label:_ , etc. But, those are not having any spellcheck issues as of now.

Please try to make it work for cases with hyphens also.

**Comment by maybeec** [comment]
Just one question, why do we need all this properties here?
Normally we generate the website / pdf with a central tool and not configuring asciidoc in each document. Is it really needed?

**Comment by sundharakanth** [comment]
> > Hi All ( @maybeec , @SchettlerKoehler , @ssarmokadam , @isandesh1986 , @sujith-mn ), I can see the following templated related non-standard words in dictionary.
> > 148 doctype 614 toc 249 idprefix 250 idseparator 516 reproducable [Looks like mis-spelling of reproducible]
> > If we have the above words, then, I think, it is okay to add the following few more words. Please confirm. (a) :imagesdir: (b) :partnums: (c) :sectnums:
> > image
> > [ FYI - @albertdb ]
> 
> That is kicking the can down the road, because we can have arbitrary attributes in a document: https://asciidoctor.org/docs/asciidoc-recommended-practices/#document-attributes-i-e-variables
> 
> > > @sundharakanth Please go ahead preparing the solution mentioned in the second option while we get more opinions on this matter. Not much work to throw away in case a different decision is finally taken.
> > 
> > 
> > @albertdb - The following regex takes care of escaping the single words like _:sectnums:_ , _:partnums:_ , _:imagesdir:_ , etc. Please confirm. [ FYI - @ssarmokadam , @isandesh1986 , @sujith-mn ]
> > image
> > **Note:** The above one does not take care of the words with hyphen in between like _:toc-title:_ , _:chapter-label:_ , etc. But, those are not having any spellcheck issues as of now.
> 
> Please try to make it work for cases with hyphens also.

image

**Comment by albertdb** [comment]
> Just one question, why do we need all this properties here? Normally we generate the website / pdf with a central tool and not configuring asciidoc in each document. Is it really needed?

I think the problem is more general. If we want to create a variable for a URL, the spellchecker will complain. For me is the same as running it against source code, completely absurd.

**Comment by sundharakanth** [comment]
Raised the PR https://github.com/devonfw/.github/pull/127 for change in spellcheck.yml

**Comment by albertdb** [comment]
@sundharakanth Any advances on this?

**Comment by sundharakanth** [comment]
> @sundharakanth Any advances on this?

@albertdb Tested the regex changes in my fork and it is working fine. Need to test negative scenarios as well. Same changes are already added to this PR in Hangar repo.

**Comment by sundharakanth** [comment]
> Hi All ( @maybeec , @SchettlerKoehler , @ssarmokadam , @isandesh1986 , @sujith-mn ), I can see the following templated related non-standard words in dictionary.
> 
> 148 doctype 614 toc 249 idprefix 250 idseparator 516 reproducable [Looks like mis-spelling of reproducible]
> 
> If we have the above words, then, I think, it is okay to add the following few more words. Please confirm. (a) :imagesdir: (b) :partnums: (c) :sectnums:
> 
> image
> 
> [ FYI - @albertdb ]

@ssarmokadam and @sujith-mn - Shall we go ahead with this option? Please confirm.

**Review by sundharakanth — APPROVED** [review]

**Comment by sundharakanth** [comment]
> @sundharakanth You have force pushed and removed changes in spellcheck.yaml
> 
> Please check.

@albertdb - Yes, thats right. To fix the issue temporarily, I added the 3 new words in dictionary and get it merged. So, removed all the .spellcheck.yml file changes. 

Now, you can merge this PR#43 which will make spellcheck workflow to execute successfully in Hangar repo.

Currently Spellcheck is passing in the branch feature/acr-credentials and failing in master. 

image

**Comment by albertdb** [comment]
@sundharakanth As I commented before about adding new attributes to dictionary:

> That is kicking the can down the road, because we can have arbitrary attributes in a document: https://asciidoctor.org/docs/asciidoc-recommended-practices/#document-attributes-i-e-variables

Please reintroduce spellcheck change if that is valid for any `:watheverVariable-with-hyphens:`

**Comment by sundharakanth** [comment]
> @sundharakanth As I commented before about adding new attributes to dictionary:
> 
> > That is kicking the can down the road, because we can have arbitrary attributes in a document: https://asciidoctor.org/docs/asciidoc-recommended-practices/#document-attributes-i-e-variables
> 
> Please reintroduce spellcheck change if that is valid for any `:watheverVariable-with-hyphens:`

@albertdb - I will discuss with Devon Core team ( @ssarmokadam , @sujith-mn , @isandesh1986 ) and get back.

**Comment by sundharakanth** [comment]
> Devon Core team ( @ssarmokadam , @sujith-mn , @isandesh1986 ) and get back

@albertdb - A JIRA Ticket is created for this requirement and Devon Core team ( @ssarmokadam , @sujith-mn , @isandesh1986 ) would get back to you on the timelines of the same.
